Added Handover Documents (#26)
ghga-de/encryption-key-store-service: .description.md

<!-- Please provide a short overview of the features of this service.-->

> This service implements an interface to extract file encryption secrects from a
> [GA4GH Crypt4GH](https://www.ga4gh.org/news/crypt4gh-a-secure-method-for-sharing-human-genetic-data/)
> encrypted file in a HashiCorp Vault and request user-specific file envelopes
> containing these secrets.

Reviewer: ```suggestion
This service implements an interface to extract file encryption secrects from a
[GA4GH Crypt4GH](https://www.ga4gh.org/news/crypt4gh-a-secure-method-for-sharing-human-genetic-data/)
encrypted file into a HashiCorp Vault and produce user-specific file envelopes
containing these secrets.
```

### API endpoints:

#### `POST /secrets`:

This endpoint takes in the first part of a crypt4gh encrypted file that contains the
file envelope and a client public key.

> It decrypts the envelope, using the clients public and GHGAs private key to get
> the encryption secret.
> It also creates a new random secret that can be used for re-encryption and stores this
> new secret in the vault.

Reviewer: ```suggestion
It decrypts the envelope, using the clients public and GHGA's private key to obtain
the original encryption secret.
Subsequently, a new random secret that can be used for re-encryption and is created 
and stored in the vault.
The original secret is *not* saved in the vault.
```

> This endpoint returns the new secret, the submitted secret, the envelope offset
> (length of the envelope) and the secret id for retrieving the new secret from the vault.

Reviewer: ```suggestion
This endpoint returns the extracted secret, the newly generated secret, the envelope offset
(length of the envelope) and the secret id which can be used to retrieve the new secret from the vault.
```

#### `GET /secrets/{secret_id}/envelopes/{client_pk}`:

> This endpoint takes in a secret_id and a client public key.

Reviewer: ```suggestion
This endpoint takes a secret_id and a client public key.
```

<!-- Please provide a short overview of the features of this service.-->

This service implements an interface to extract file encryption secrects from a
[GA4GH Crypt4GH](https://www.ga4gh.org/news/crypt4gh-a-secure-method-for-sharing-human-genetic-data/)
encrypted file in a HashiCorp Vault and request user-specific file envelopes
containing these secrets.

> The Vault is owned and controlled by this service, no other service has access to it.

Reviewer 2: This is a integration detail and not enforced in the code base, maybe better to remove.
```suggestion
```
